### DISCUSSION TITLE
Filament 3.0 : How to Encrypt the sensitive data from the response of Request Payload

### DISCUSSION THREAD

**Discussion by rzortiz11** [filamentphp/filament#12818] [Help]
### Package

Form builder

### Package Version

v3.0.0.0

### How can we help you?

Hi, can some one tell me if it is possible as of the moment to encrypt a data that was being sent as a response to the client-side using filament 3.0? if it is not possible can you provide me a solution? 

I have done testing encrypting a a field but using a blade.php for example by replacing the filament Login to blade.php Login  and javascript but can filament handle it already? 

The response between from mutateFormDataBeforeFill and mutateFormDataBeforeSave seems the one displaying the response or is it the action button?

EDITED : or One more thing can the responsePayload be removed or hidden so that senstive data cant be view in the client side?

Using Laravel 11 and Filament 3.0 for Admin Page.

![image](https://github.com/filamentphp/filament/assets/156640289/3184a1e8-901e-4678-8f1e-329b821c36e8)
![image](https://github.com/filamentphp/filament/assets/156640289/64408a78-d402-45d4-aa60-63e9dbd741c0)

**Comment by ekrist1** [comment]
You should be able to protect model attributes using the $hidden [attribute](https://laravel.com/docs/11.x/eloquent-serialization#hiding-attributes-from-json) in your model. 

I´m using the mutateFormDataBeforeFill in the edit/list page to avoid leaking sensitive data to the frontend:

```
 protected function mutateFormDataBeforeFill(array $data): array
    {
        if (Auth::user()->can('sensitiveData', [Employee::class, $data])) {
            return $data;
        }
        unset($data['x']);
        unset($data['y']);
        unset($data['z']);
    
        return $data;
    }
}
```

**Reply by rzortiz11** [reply]
@ekrist1 Thank you for the feed back, will that also work on the response? since livewire/update return components attributes aswell.

The simplest work around i did to encrypt the payload of data and to hide or customize the response was to create a livewire component, re-use the filament form for the input field to make it easier, and  add a submit button + Javascript to make the client side encrypt the payload and have the backend decrypt the payload and customize the response. 

But i dont like the approach i did, it is too much of a hassle if you have so many forms.

**Reply by ekrist1** [reply]
I haven't done extensive testing, but from what I can see, the field which is unset is not present in the payload when submiting a form.

**Comment by sakthimanikandan558** [comment]
> ### Package
> Form builder
> 
> ### Package Version
> v3.0.0.0
> 
> ### How can we help you?
> Hi, can some one tell me if it is possible as of the moment to encrypt a data that was being sent as a response to the client-side using filament 3.0? if it is not possible can you provide me a solution?
> 
> I have done testing encrypting a a field but using a blade.php for example by replacing the filament Login to blade.php Login and javascript but can filament handle it already?
> 
> The response between from mutateFormDataBeforeFill and mutateFormDataBeforeSave seems the one displaying the response or is it the action button?
> 
> EDITED : or One more thing can the responsePayload be removed or hidden so that senstive data cant be view in the client side?
> 
> Using Laravel 11 and Filament 3.0 for Admin Page.
> 
> ![image](https://private-user-images.githubusercontent.com/156640289/331115370-3184a1e8-901e-4678-8f1e-329b821c36e8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzk3NzEzODgsIm5iZiI6MTc3OTc3MTA4OCwicGF0aCI6Ii8xNTY2NDAyODkvMzMxMTE1MzcwLTMxODRhMWU4LTkwMWUtNDY3OC04ZjFlLTMyOWI4MjFjMzZlOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyNlQwNDUxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjA3YzQwN2U0NDA1ZDQ3M2FhYjI4NmNjYTBiYWU2YzU1ODY2MWE0MmFjYzlkOWQ3OTMyNDFmYWUzMTdiNGZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.oUJUPed_GFJlJR_f2j6MJJsBtfAX6zd-R-fjn8oFL1U) ![image](https://private-user-images.githubusercontent.com/156640289/331115476-64408a78-d402-45d4-aa60-63e9dbd741c0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzk3NzEzODgsIm5iZiI6MTc3OTc3MTA4OCwicGF0aCI6Ii8xNTY2NDAyODkvMzMxMTE1NDc2LTY0NDA4YTc4LWQ0MDItNDVkNC1hYTYwLTYzZTlkYmQ3NDFjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyNlQwNDUxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNThjMDljODcyZWJiZDhlMTBjYzk2NjhlMWQyOGQ2MDA1N2YzMzNmMTdlZDc3Mjc1ODQ1YTQ1ZjBlMWM2OWQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.7knJieCLkY7ovek0dUTpysgDu17sLHsjvmvyY4xiYdk)

Can you give me your code snippet.?
